components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/claims/BFSIDefaultClaimProvider.java

```diff
@@ -0,0 +1,151 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.carbon.identity.openidconnect.model.Constants.JWT_PART_DELIMITER;
+import static org.wso2.carbon.identity.openidconnect.model.Constants.NUMBER_OF_PARTS_IN_JWE;
+
+/**
+ * Default BFSI specific claim provider implementation.
+ */
+public class BFSIDefaultClaimProvider extends BFSIClaimProvider {
+
+    private static final Log log = LogFactory.getLog(BFSIDefaultClaimProvider.class);
+
+    @Override
+    public Map<String, Object> getAdditionalClaims(OAuthAuthzReqMessageContext authAuthzReqMessageContext,
+                                                   OAuth2AuthorizeRespDTO authorizeRespDTO)
+            throws IdentityOAuth2Exception {
+
+        Map<String, Object> claims = new HashMap<>();
+        String[] cachedRequests = null;
+        final String sessionDataKey = authAuthzReqMessageContext.getAuthorizationReqDTO().getSessionDataKey();
+        if (StringUtils.isNotBlank(sessionDataKey)) {
+            cachedRequests = SessionDataCache.getInstance()
+                    .getValueFromCache(new SessionDataCacheKey(sessionDataKey)).getParamMap().get("request");
+        }
+        if (cachedRequests != null && !(cachedRequests[0].split(JWT_PART_DELIMITER).length == NUMBER_OF_PARTS_IN_JWE)) {
+            JSONObject requestBody = getRequestBodyFromCache(cachedRequests);
+
+            /* State is an optional parameter, so the authorization server must successfully authenticate and
+             * must NOT return state nor s_hash. (FAPI1-ADV-5.2.2.1-5)
+             */
+            final String state = requestBody.getAsString(OAuthConstants.OAuth20Params.STATE);
+            if (StringUtils.isNotEmpty(state)) {
+                claims.put(IdentityCommonConstants.S_HASH, IdentityCommonUtils.getHashValue(state, null));
+            } else {
+                // state is empty, removing state from cache too
+                removeStateFromCache(sessionDataKey);
+            }
+        }
+        final String responseType = authAuthzReqMessageContext.getAuthorizationReqDTO().getResponseType();
+        avoidSettingATHash(responseType, authorizeRespDTO, claims);
+
+        return claims;
+
+    }
+
+    @Override
+    public Map<String, Object> getAdditionalClaims(OAuthTokenReqMessageContext tokenReqMessageContext,
+                                                   OAuth2AccessTokenRespDTO tokenRespDTO)
+            throws IdentityOAuth2Exception {
+
+        return new HashMap<>();
+    }
+
+    /**
+     * If response_type value is not 'code id_token token', avoid setting at_hash claim to the authorization
+     * endpoint id_token as it is OPTIONAL (OIDCC-3.3.2.11).
+     *
+     * @param responseType     requested auth response_type
+     * @param authorizeRespDTO authorizeRespDTO
+     * @param claims           returning claims map
+     */
+    private void avoidSettingATHash(String responseType, OAuth2AuthorizeRespDTO authorizeRespDTO,
```

Comment:
Shall we use a more clear method name? eg: "setATHashAsNull"?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/fc5b5f2d081c4d52de5072aa80c4cafa18e006b3